runtime/query.go

```diff
@@ -291,7 +291,11 @@ func parseMessage(msgDescriptor protoreflect.MessageDescriptor, value string) (p
 		if err != nil {
 			return protoreflect.Value{}, err
 		}
-		msg = timestamppb.New(t)
+		timestamp := timestamppb.New(t)
+		if ok := timestamp.IsValid(); !ok {
```

Comment:
Nit: now that I look at [the docs for `timestamppb`](https://pkg.go.dev/google.golang.org/protobuf/types/known/timestamppb#Timestamp.CheckValid), perhaps we should use `timestamp.CheckValid()` instead so that we can tell the user why it was an invalid timestamp?

Reply:
Agreed. Will do

Reply:
The error returned from `timestamp.CheckValid` wasn't very friendly and was difficult to test. Looks like this was by [design](https://github.com/protocolbuffers/protobuf-go/blob/30f628eeb303f2c29be7a381bf78aa3e3aabd317/internal/errors/errors.go#L27)

But the only possible error here is timestamp before 0001-01-01 so just returned that message

Reply:
Alright fair enough